src/main/java/Handler/FileHandler.java

```diff
@@ -11,29 +13,39 @@ public Response generateResponse(Request request) {
      
     switch (request.getMethod()) { 
       case "GET":  
-        return buildGETResponse(); 
+        return buildGetResponse(); 
       default:  
         return new Response.Builder(HttpStatusCode.METHOD_NOT_ALLOWED) 
                            .build(); 
     } 
   }
 
-  private Response buildGETResponse() { 
+  private Response buildGetResponse() {
     int statusCode = determineStatusCode();
-    String messageBody = createMessageBody(statusCode);
-     
-    return new Response.Builder(statusCode) 
-                       .messageBody(messageBody) 
-                       .build(); 
-  } 
+    String messageBody = createMessageBody();
+    int contentLength = ResponseHeader.determineContentLength(messageBody);
+    String contentType = determineContentType();
+    return new Response.Builder(statusCode)
+                   .messageBody(messageBody)
+                   .contentLength(contentLength)
+                   .contentType(contentType)
+                   .build();
+  }
 
   private int determineStatusCode() {
-    return store.existsInStore(this.uri) ? HttpStatusCode.OK : HttpStatusCode.NOT_FOUND;
+    return this.store.existsInStore(this.uri) ? HttpStatusCode.OK : HttpStatusCode.NOT_FOUND;
   }
 
-  private String createMessageBody(int statusCode) {
-    String emptyMessageBody = "";
-    return (statusCode == HttpStatusCode.OK) ? store.read(this.uri) : emptyMessageBody;
+  private String createMessageBody() {
+    return this.store.existsInStore(this.uri) ? store.read(this.uri) : buildNotFoundMessage();
```

Comment:
not a big deal, but is there a way you can call `existsInStore` once to build the response instead of twice?